js/views/RegistrationPages.js

```diff
@@ -123,7 +123,8 @@ const RegistrationPages = React.createClass ({
    */
   beforePageChange(currentPage, nextPage) {
     const shouldContinue = this.validatePage();
-    if (shouldContinue || nextPage < currentPage) {
+    if (shouldContinue || nextPage >= 0 && nextPage < currentPage) {
```

Comment:
Truth tables are a bit off on this, ex:
```
false || true && true
> true
```

So its possible to navigate forward even if should continue is false (validation error)